packages/website/src/widgets/citation/ui/search/comparative-search.tsx

```diff
@@ -42,48 +46,75 @@ export default function ComparativeSearch(props: ComparativeSearchProps) {
     );
   }
 
-  return (
-    <>
-      <CitationExplorerTitle onClose={onClose}>Compare Mode</CitationExplorerTitle>
+  useEffect(() => {
+    if (regionType) {
+      setCurrentStep(1);
+    } else if (region1) {
+      setCurrentStep(2);
+    }
+  }, [regionType, region1]);
 
-      <CitationExplorerSection>
-        <CitationExplorerSectionTitle>Select One Region Type to Compare</CitationExplorerSectionTitle>
+  const regionTypeTitle = (
+    <CitationExplorerSectionTitle>Select One Region Type to Compare</CitationExplorerSectionTitle>
+  );
 
-        <RegionTypesSelection onChange={onRegionTypeSelect} />
+  const firstRegionTitleContainer = (
+    <div className="flex w-[420px] justify-between">
+      <CitationExplorerSectionTitle>First Region</CitationExplorerSectionTitle>
+      {region1 && (
+        <Button variant={ButtonVariant.text} onClick={() => setDataFocus(1)}>
+          See Data 1
+        </Button>
+      )}
+    </div>
+  );
 
-        <div className="flex justify-between">
-          <CitationExplorerSectionTitle>First Region</CitationExplorerSectionTitle>
-          {region1 && (
-            <Button variant={ButtonVariant.text} onClick={() => setDataFocus(1)}>
-              See Data 1
-            </Button>
-          )}
-        </div>
+  const secondRegionTitleContainer = (
+    <div className="flex w-[420px] justify-between">
+      <CitationExplorerSectionTitle>Second Region</CitationExplorerSectionTitle>
+      {region2 && (
+        <Button variant={ButtonVariant.text} onClick={() => setDataFocus(2)}>
```

Comment:
Both onclick navigation is breaking the app

Reply:
@gibsonliketheguitar What error do you get?

Reply:
I couldn't replicate any error, but I went ahead and fixed an issue where there the inputs (region type, first region, and second region) wouldn't repopulate based on the states after clicking "see data 1" (or "see data 2" or the exit "X") and returning back to the "Compare Mode" page. Hopefully that fixes it, but let me know and I can take another look at it!

Reply:
I was getting 
```Unexpected Application Error!
Rendered fewer hooks than expected. This may be caused by an accidental early return statement.```

in dev mode. I suspect is related to the useEffect